### PR TITLE
Changes for #SR-9413:Swift 5.0 test failures on PowerPC

### PR DESCRIPTION
**Pull request by sarveshtamba**
Changes for #SR-9413:Swift 5.0 test failures on PowerPC

### CODE REVIEW

**Comment by sarveshtamba** [comment]
@compnerd For some reason had put an unsupported tag in "validation-test/stdlib/String.swift" during my testing and missed reverting it. Nevertheless, have reverted the changes as it passes without any modifications on ppc64le. Please let me know your suggestions on the others. Thanks!

**Comment by compnerd** [comment]
Some of them make sense, as in sure, tsan may not be available on ppc64le, but most of the others should be passing.  I think that you need to look into these rather than just disabling them.  Witness  table handling must work for example.

**Comment by sarveshtamba** [comment]
As mentioned in SR-9474, I checked the link under section "More Information" - https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual

**Supported Platforms:
TSan is supported on Linux x86_64, tested on Ubuntu 12.04.**

Hence I disabled Sanitizers/tsan.swift and validation-test/Sanitizers/witness_table_lookup.swift(which makes us of  "// REQUIRES: tsan_runtime") for PowerPC64LE and tested further.

**Comment by compnerd** [comment]
Right, I was saying that disabling that particular test makes sense, but most of the other ones don't.  However, I did miss that it has a `REQUIRES: tsan_runtime`.  I think that you should be adjusting the feature in lit rather than adding an `UNSUPPORTED: powerpc64le` there (basically, that test shouldn't have a change at all).

**Comment by sarveshtamba** [comment]
@compnerd , I am relatively new to this testing and Swift in general, can you please provide any inputs on "adjusting the feature in lit"? TIA.

**Comment by compnerd** [comment]
I would recommend looking at the documentation on lit in LLVM.

**Comment by sarveshtamba** [comment]
Hi @jrose,

I tried building the toolchain with the --test option on linux-x86_64 platform. Only the following tests fail on linux-x86_64 platform.

4 warning(s) in tests.
Testing Time: 1645.16s
********************
Failing Tests (6):
    Swift(linux-x86_64) :: remote-run/upload-stderr.test-sh
    Swift(linux-x86_64) :: Runtime/linux-fatal-backtrace.swift
    Swift(linux-x86_64) :: ParseableInterface/ModuleCache/force-module-loading-mode-archs.swift
    Swift(linux-x86_64) :: ParseableInterface/ModuleCache/force-module-loading-mode-framework.swift
    Swift(linux-x86_64) :: ParseableInterface/ModuleCache/force-module-loading-mode.swift
    Swift(linux-x86_64) :: Index/Store/output-failure.swift

  Expected Passes    : 10279
  Expected Failures  : 45
  Unsupported Tests  : 1255
  Unexpected Failures: 6

Any suggestions would greatly help. Thanks in advance.

**Comment by jrose-apple** [comment]
It's not expected that any of those fail on x86_64 Linux—or at least not on one of the supported Ubuntu versions. You can check https://ci.swift.org to see that they're all passing. It'll take more investigation to determine why they're failing for you, and that probably doesn't belong on this PR.

File: test/ClangImporter/inlinable_bitfields.swift [hunk 1]
```diff
@@ -11,4 +10,4 @@ public func g(_ m: MM) -> UInt32 {
 
 // Just make sure this is a definition and not a declaration...
 
-// CHECK: define internal i32 @"$ModRM$rm$getter"
+// CHECK: define internal zeroext i32 @"$ModRM$rm$getter"
```
Comment (jrose-apple) [inline test/ClangImporter/inlinable_bitfields.swift]: Won't this break platforms that aren't using `zeroext`?

You can use a FileCheck regex to handle this: `{{( zeroext)?}}`.

**Review by jrose-apple — COMMENTED** [review]
Thanks. I'm still not 100% convinced it's interesting to disable these tests rather than just leave them failing, but I like the XFAIL a lot better.

File: test/ClangImporter/inlinable_bitfields.swift (hunk 1 above)
Comment (sarveshtamba) [reply test/ClangImporter/inlinable_bitfields.swift]: Updated to use {{( zeroext)?}}. Pls review. Thanks!

**Comment by sarveshtamba** [comment]
Hi @jrose,
I have put probable fixes for the following test cases which were earlier failing on PPC64LE and marked as UNSUPPPORTED in this PR#21541.
FIX:
test/IRGen/pic.swift
test/ClangImporter/inlinable_bitfields.swift
test/IRGen/objc_simd.sil
test/Constraints/diagnostics.swift
XFAIL:
test/IRGen/class_resilience_objc.swift (marked as XFAIL for CPU=armv7k too)
All other test cases which were earlier marked as UNSUPPPORTED and not yet fixed are now marked as XFAIL instead, as per suggestions above.
However, I have still left the sanitizers related tests(tsan.swift, fuzzer.swift, witness_table_lookup.swift) as UNSUPPPORTED due to reasons in the respective SR's mentioned above.
Please let me know your thoughts on this.

**Comment by jrose-apple** [comment]
That sounds reasonable to me! "Sanitizers are unsupported [for now?] on PowerPC; everything else ought to work but doesn't yet for some reason."

@swift-ci Please test

**Review by jrose-apple — APPROVED** [review]

**Pull request merged**